api/v1/controllers/samples.js

```diff
@@ -150,13 +151,25 @@ module.exports = {
     const resultObj = { reqStartTime: new Date() };
     const sampleQueryBody = req.swagger.params.queryBody.value;
 
-    return u.getUserNameFromToken(req,
-      featureToggles.isFeatureEnabled('enforceWritePermission'))
-    .then((userName) => {
+    return authUtils.getUser(req)
+    .then((user) => {
       if (sampleQueryBody.relatedLinks) {
         u.checkDuplicateRLinks(sampleQueryBody.relatedLinks);
       }
 
+      if (featureToggles.isFeatureEnabled('returnCreatedBy') && user) {
+        sampleQueryBody.createdBy = user.id;
+      }
+
+      let userName = '';
+      if (!featureToggles.isFeatureEnabled('enforceWritePermission')) {
+        userName = true;
```

Comment:
we will be removing the "enforceWritePermission" probably the sprint starting 5/15. I think your changes will be much cleaner if you do it without checking for this flag.